core/src/main/java/com/oath/oak/ValueUtils.java

```diff
@@ -0,0 +1,384 @@
+
+    static void unsafeBufferToIntArrayCopy(ByteBuffer bb, int srcPosition, int[] dstArray, int countInts) {
+        UnsafeUtils.unsafeCopyBufferToIntArray(bb, srcPosition, dstArray, countInts);
+    }
+
+    /**
+     * Used only in OakRValueBuffer
+     *
+     * @param bb          the value Byte Buffer (including the header)
+     * @param transformer value deserializer
+     * @param <T>         the type of {@code transformer}'s output
+     * @return {@code null} if the value is deleted or was moved.
+     * The value written off-heap is returned otherwise.
+     */
+    static <T> T transform(ByteBuffer bb, Function<ByteBuffer, T> transformer) {
+        if (lockRead(bb) != SUCCESS) {
+            return null;
+        }
+
+        T transformation = transformer.apply(getValueByteBufferNoHeader(bb).asReadOnlyBuffer());
+        unlockRead(bb);
+        return transformation;
+    }
+
+    /**
+     * Used to try and read a value off-heap
+     *
+     * @param s           the value Slice (including the header)
+     * @param transformer value deserializer
+     * @param <T>         the type of {@code transformer}'s output
+     * @return {@code SUCCESS} if the value was read successfully,
+     * {@code FAILURE} if the value is deleted,
+     * {@code RETRY} if the value was moved.
+     * In case of {@code SUCCESS}, the value of the returned entry is the read value, otherwise, the value is {@code
+     * null}.
+     */
+    static <T> Result<T> transform(Slice s, Function<ByteBuffer, T> transformer) {
+        ByteBuffer bb = s.getByteBuffer();
+        ValueResult res = lockRead(bb);
+        if (res != SUCCESS) {
+            return Result.withFlag(res);
+        }
+
+        T transformation = transformer.apply(getValueByteBufferNoHeader(bb).asReadOnlyBuffer());
+        unlockRead(bb);
+        return Result.withValue(transformation);
+    }
+
+    /**
+     * Replaces the value written in the Slice referenced by {@code lookup} with {@code newVal}.
+     * {@code chuck} is used iff {@code newValue} takes more space than the old value.
+     * If the value moves, the old slice is marked as moved and freed.
+     *
+     * @param chunk         the chunk with the entry to which the value is linked to
+     * @param lookUp        has the Slice of the value and the entry index
+     * @param newVal        the new value to write
+     * @param serializer    value serializer
+     * @param memoryManager the memory manager to free a slice with is not needed after the value moved
+     * @param <V>           the type of the value
+     * @return {@code SUCCESS} if the value was written off-heap successfully,
+     * {@code FAILURE} if the value is deleted (cannot be overwritten),
+     * {@code RETRY} if the value was moved, or if the chuck is frozen/released (prevents the moving of the value).
+     */
+    public static <V> ValueResult put(Chunk<?, V> chunk, Chunk.LookUp lookUp, V newVal, OakSerializer<V> serializer,
```

Comment:
Is it used only for putIfPresent? Right? Please comment

Reply:
It is not used, so I deleted it.